src/ape/contracts/base.py

```diff
@@ -1628,6 +1628,12 @@ def deploy(self, *args, publish: bool = False, **kwargs) -> ContractInstance:
         Returns:
             :class:`~ape.contracts.base.ContractInstance`
         """
+        # TODO: Not possible method not available
+
+        if "sender" in kwargs:
+            possible_address = kwargs["sender"].get_deployment_address()
```

Comment:
This would be a more failsafe way of doing this per @antazoey's comment
```suggestion
        if isinstance(sender := kwargs.get("sender"), AccountAPI):
            possible_address = sender.get_deployment_address()
```